src/components/CharactersPage/CharacterForm/CharacterForm.js

```diff
@@ -62,7 +62,18 @@ const CharacterForm = ({ onClose }) => {
     return params[0] === formModeEnum.edit
   }, [params])
 
-  const character = useSelector(characterSelectors.character(params[1]))
+  const characterCacheId = client.cache.identify({
+    __typename: "Person",
+    id: params[1],
+  })
+
+  const character = client.readFragment({
+    id: characterCacheId,
+    fragment: GET_A_CHARACTER,
+  })
```

Comment:
You should not be using fragments here, find a way to do this with the query hooks provided by apollo